cob_twist_controller/include/cob_twist_controller/cob_twist_controller.h

```diff
@@ -147,8 +153,21 @@ class CobTwistController
 	void twist_stamped_cb(const geometry_msgs::TwistStamped::ConstPtr& msg);
 	void solve_twist(KDL::Twist twist);
 	
-	KDL::JntArray normalize_velocities(KDL::JntArray q_dot_ik);
-	
+	/**
+	 * Needed for limiting the base velocities
+	 * @param The inverse kinematics solution for the joint velocity vector
+	 * @return The normalized joint velocity vector
+	 */
+	KDL::JntArray normalize_velocities(const KDL::JntArray &q_dot_ik) const;
```

Comment:
See latest latest commit 5703cdc (20th April '15).
New structure for enforcing limits -> new classes which can be concatenated and made independent.

Added keep_direction parameter to cfg to choose between all and individual.